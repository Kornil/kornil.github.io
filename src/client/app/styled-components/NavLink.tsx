import styled from "@emotion/styled";
import { NavLink as Link } from "react-router-dom";

const NavLink = styled(Link)`
  color: inherit;
  text-decoration: none;
  letter-spacing: 4px;

  :hover {
    text-decoration: underline;
  }

  &.active {
    text-decoration: line-through;
  }
`;

export default NavLink;
