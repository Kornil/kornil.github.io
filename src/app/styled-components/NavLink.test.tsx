import { shallow } from "enzyme";
import React from "react";

import NavLink from "./NavLink";

test("it works", () => {
  const wrapper = shallow(<NavLink to="/random" />);
  expect(wrapper).toMatchSnapshot();
});
