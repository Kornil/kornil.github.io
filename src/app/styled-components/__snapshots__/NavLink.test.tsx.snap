// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.emotion-0 {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 28px;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0.active {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

<NavLink
  activeClassName="active"
  aria-current="page"
  className="emotion-0 emotion-1"
  to="/random"
/>
`;
